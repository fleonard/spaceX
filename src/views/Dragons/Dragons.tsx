import React from "react";

import { Typography } from "@material-ui/core";

import { useSpaceX } from "../../hooks/useSpaceX";

import Spinner from "../../components/Spinner/Spinner";
import Error from "../../components/Error/Error";
import ListItem from "../../components/ListItem/ListItem";

import { ColumnContainer, ListWrapper } from "../../globalStyles";

const Dragons: React.FC = () => {
  const { data, error, status } = useSpaceX("/dragons");

  return (
    <ColumnContainer>
      <Typography variant="h4">Dragons</Typography>

      {status === "fetching" && <Spinner />}

      {status === "error" && <Error text={error} />}

      {status === "fetched" && (
        <ListWrapper>
          {data.map((dragon) => (
            <ListItem key={dragon.id} item={dragon} />
          ))}
        </ListWrapper>
      )}
    </ColumnContainer>
  );
};

export default Dragons;
