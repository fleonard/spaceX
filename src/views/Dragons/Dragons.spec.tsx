import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import "jest-styled-components";

import Dragons from "./Dragons";

describe("Dragon Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Dragons />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the dragon list", async () => {
    const { getByText, getAllByTestId } = render(
      <BrowserRouter>
        <Dragons />
      </BrowserRouter>
    );

    await waitFor(() => expect(getAllByTestId("dragon")).toHaveLength(2));
    await waitFor(() => expect(getByText("Dragon 1")).toBeDefined());
    await waitFor(() => expect(getByText("Dragon 2")).toBeDefined());
  });
});
