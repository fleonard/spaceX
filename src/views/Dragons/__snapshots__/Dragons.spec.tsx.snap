// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon Component renders correctly 1`] = `
.c1 {
  display: grid;
  padding: 30px 0;
  place-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px 20px;
}

.c2.c2 {
  color: #000;
}

<div
  class="c0"
>
  <h4
    class="MuiTypography-root MuiTypography-h4"
  >
    Dragons
  </h4>
  <div
    class="c1"
  >
    <div
      class="MuiCircularProgress-root c2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
